src/transports/ipc/bipc.c

```diff
@@ -168,6 +168,11 @@ static void nn_bipc_destroy (struct nn_epbase *self)
 static void nn_bipc_shutdown (struct nn_fsm *self, int src, int type,
     void *srcptr)
 {
+#if !defined NN_HAVE_WINDOWS
```

Comment:
I used the same test as the matching [unlinking code below](https://github.com/Snaipe/nanomsg/blob/ipc-unlink/src/transports/ipc/bipc.c#L415-L429) to be consistent, but you're right: it should definitely be inside a feature test rather than a platform test.

There is currently no way to check if unix domain sockets are supported, other than supposing that all platforms other than windows support them. Maybe a separate commit should address this specific issue, but checking for availability of domain socket is not as trivial as checking for `AF_UNIX`, because windows actually provide a definition of it without using it in practice.